### ISSUE TITLE
darc exceptions

### ISSUE THREAD

**Issue by tmat** [dotnet/arcade#1602]
Sync dotnet/symreader to commit 26a5d2426f77a99d3196d80b2e7322829d20b60f.

```
>powershell .\eng\common\darc-init.ps1
Tool 'microsoft.dotnet.darc' (version '1.0.0-beta.18616.5') was successfully installed.
```

```
>darc get-dependency-graph
fail: Microsoft.DotNet.Darc.Operations.Operation[0]
      Something failed while trying the fetch the dependencies of repo 'https://github.com/dotnet/arcade' at sha '8694eb83a2c13057a8da3c80f829363de4dc8c9c'
System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at Microsoft.DotNet.DarcLib.DependencyGraph.GetRepoPath(DependencyDetail dependency, IEnumerable`1 remotesMap, String reposFolder, ILogger logger) in /_/src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs:line 103
   at Microsoft.DotNet.DarcLib.DependencyGraph.GetDependenciesAsync(DarcSettings darcSettings, Boolean remote, ILogger logger, DependencyGraphNode node, IEnumerable`1 remotesMap, String reposFolder, String testPath) in /_/src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs:line 190
fail: Microsoft.DotNet.Darc.Operations.Operation[0]
      Something failed while getting the dependency graph.
System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at Microsoft.DotNet.DarcLib.DependencyGraph.GetRepoPath(DependencyDetail dependency, IEnumerable`1 remotesMap, String reposFolder, ILogger logger) in /_/src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs:line 103
   at Microsoft.DotNet.DarcLib.DependencyGraph.GetDependenciesAsync(DarcSettings darcSettings, Boolean remote, ILogger logger, DependencyGraphNode node, IEnumerable`1 remotesMap, String reposFolder, String testPath) in /_/src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs:line 190
   at Microsoft.DotNet.DarcLib.DependencyGraph.GetDependencyGraphAsync(DarcSettings darcSettings, DependencyDetail dependency, Boolean remote, ILogger logger, String reposFolder, IEnumerable`1 remotesMap, String testPath) in /_/src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs:line 56
   at Microsoft.DotNet.Darc.Operations.GetDependencyGraphOperation.AddNodeToGraphAsync(DarcSettings darcSettings, DependencyDetail dependency, List`1 graph) in /_/src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyGraphOperation.cs:line 125
   at Microsoft.DotNet.Darc.Operations.GetDependencyGraphOperation.CreateGraphAsync(IEnumerable`1 dependencies, DarcSettings darcSettings) in /_/src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyGraphOperation.cs:line 98
   at Microsoft.DotNet.Darc.Operations.GetDependencyGraphOperation.ExecuteAsync() in /_/src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyGraphOperation.cs:line 57
```

```
>darc verify
D:\symreader>darc verify
fail: Microsoft.DotNet.Darc.Operations.Operation[0]
      Error: Failed to verify repository dependency state.
System.Xml.XPath.XPathException: Expression must evaluate to a node-set.
   at MS.Internal.Xml.XPath.XPathParser.ParseNodeTest(AstNode qyInput, AxisType axisType, XPathNodeType nodeType)
   at MS.Internal.Xml.XPath.XPathParser.ParseStep(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseRelativeLocationPath(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseLocationPath(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParsePathExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseUnionExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseUnaryExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseMultiplicativeExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseAdditiveExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseRelationalExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseEqualityExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseAndExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseOrExpr(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseExpression(AstNode qyInput)
   at MS.Internal.Xml.XPath.XPathParser.ParseXPathExpression(String xpathExpression)
   at System.Xml.XPath.XPathExpression.Compile(String xpath, IXmlNamespaceResolver nsResolver)
   at System.Xml.XPath.XPathNavigator.Select(String xpath)
   at System.Xml.XmlNode.SelectNodes(String xpath)
   at Microsoft.DotNet.DarcLib.GitFileManager.VerifyMatchingVersionProps(IEnumerable`1 dependencies, XmlDocument versionProps, Task`1& utilizedDependencies) in /_/src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs:line 611
   at Microsoft.DotNet.DarcLib.GitFileManager.Verify(String repo, String branch) in /_/src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs:line 530
   at Microsoft.DotNet.Darc.Operations.VerifyOperation.ExecuteAsync() in /_/src/Microsoft.DotNet.Darc/src/Darc/Operations/VerifyOperation.cs:line 35
```

**Comment by mmitche** [comment]
See https://github.com/dotnet/arcade/issues/1515 for get-dependency graph issue.

I'll look into the verify case.

**Comment by mmitche** [comment]
Fixed by #1603
